### ISSUE TITLE
Success/alert notification link text fails contrast check

### ISSUE THREAD

**Issue by chosak** [cfpb/design-system#1955]
**Describe the bug**
Our "success" and "error" notification link texts fail the Lighthouse accessibility check for foreground / background contrast.

**To Reproduce**
1. Go to https://cfpb.github.io/design-system/components/alerts
2. In Chrome developer tools, run a Lighthouse audit

**Expected behavior**
Link text contrast should pass accessibility checks.

**Screenshots**
<img width="806" alt="image" src="https://github.com/cfpb/design-system/assets/654645/cd78b9cd-6d60-4774-a570-09a0f1585f56">

**Additional context**
Discovered via automated GitHub Action [here](https://github.com/cfpb/design-system/actions/runs/8714079706/job/23903674302). See failing Lighthouse report [here](https://storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/1713313217316-57850.report.html).

Success text is #0072ce (pacific) on #f0f8eb (green-10), which is 4.49:1.
Error text is #0072ce (pacific) on #fbefec (red-10), which is 4.34:1.

WCAG AA requirement is 4.5:1.

See previous related report at #1943; @anselmbradford tried to fix in #1944 but it seems this broke somewhere, maybe in #1919?

@sonnakim for awareness

**Comment by anselmbradford** [comment]
I see the issue. There's a group of notification modifiers 

```
&__success,
&__warning,
&__error {
    & a {
```

…that slipped through the cracks and didn't get updated from `__` to `--`.
I'll fix this soon.
